### PR TITLE
fixed calcualtions and added Jammat time

### PR DESCRIPTION
**Pull request by bytepharaoh**
Summary: fix tomorrow reminder math so minute offsets subtract the full duration from midnight and add a Jama’at reminder that fires after each prayer (20 m after Fajr, 10 m after the others) plus persistence for its last message/timestamp.

Details: extend the reminder domain with ReminderTypeJamaat/state, seed new chats & migration with the default delays, include Jama’at reminder in the scheduler, and keep the database in sync.

Testing: go test ./... (includes new domain helpers tests and reminder logic coverage).

### CODE REVIEW

File: _scripts/migrate/main.go [hunk 1]
```diff
@@ -52,6 +52,12 @@ type JamaatDelayConfig struct {
 type JamaatConfig struct {
 	Enabled bool               `json:"enabled"`
 	Delay   *JamaatDelayConfig `json:"delay"`
+	State   *JamaatState       `json:"state"`
+}
+
+type JamaatState struct {
```
Comment (escalopa) [inline _scripts/migrate/main.go]: Can you explain why we need this struct?
It has the same fields as `ReminderConfig`

File: domain/duration_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	}
+}
+
+func TestDuration_UnmarshalJSON_Invalid(t *testing.T) {
+	var d Duration
+	err := json.Unmarshal([]byte(`"not valid"`), &d)
+	if err == nil {
+		t.Fatalf("expected error for invalid duration")
+	}
+}
```
Comment (escalopa) [inline domain/duration_test.go]: It's better to write tests using array struct approach instead
Kinda the standard way of writing tests in Go (most common)
```
tests := []struct{
// fields that will be used in the tests
}{
{}, // test case 1
{}, // test case 2
}
```
like `remind_test.go`

File: domain/prayer_test.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	if got.Year() != 2024 || got.Month() != time.December || got.Day() != 31 {
+		t.Fatalf("unexpected date: %v", got)
+	}
+	if got.Location() != time.UTC {
+		t.Fatalf("expected UTC location")
+	}
+	if got.Hour() != 0 || got.Minute() != 0 || got.Second() != 0 {
+		t.Fatalf("expected midnight UTC, got %v", got)
+	}
+}
```
Comment (escalopa) [inline domain/prayer_test.go:52]: Same comment as https://github.com/escalopa/prayer-bot/pull/19/changes#r3047657828

**Comment by bytepharaoh** [comment]
**Today/Tomorrow reminders now honor minute offsets**
The “tomorrow preview” trigger now subtracts the full configured duration from midnight rather than truncating to whole hours. That means if you set remind > tomorrow +2h30m, the reminder fires at 21:30 instead of being forced to 21:00, which matches user expectations for minute-level adjustments.
**Added Jama’at reminders with per-prayer delays**
We introduced a Jama’at reminder that fires once after each main prayer (Fajr defaults to 20 m, all others to 10 m). The reminder uses the same reminder-state structure, stores the last sent message/last_at, and deletes old notices so chats don’t accumulate stale polls or messages. It’s useful for notifying groups when Jama’at is about to begin, and it ties into the UI where admins can enable/adjust Jama’at delays with the existing keyboards.
**Persisted Jama’at reminder state**
The dispatcher now seeds new chats (and the migration helper) with the Jama’at delay defaults and state, and the reminder service updates that state every time a Jama’at notice is sent. Reusing ReminderConfig keeps the schema consistent while still tracking message IDs and timestamps.
**Added unit tests for the new logic**
Domain helpers (duration formatting/parsing and date creation) now have table-driven coverage, and the reminder logic tests cover “tomorrow” deadlines plus Jama’at trigger edge cases (disabled reminders, zero delays, already-sent). That proves the behavior before deployment.
**Example use case**
A group enables Jama’at reminders and leaves the default 10‑minute offset. After Maghrib time is stored for the day, the scheduler fires the new Jama’at reminder 10 minutes later, deletes the previous Jama’at message, and sends a Markdown message like 🕌 Jama'at after 10m. If the group adjusts Jama’at delay to 15 minutes, the same flow still works because the delay is stored per prayer and respected before the reminder sends again.

**Comment by escalopa** [comment]
The PR isn't actual anymore, the bot is migrating to a newer version (global) soon, so I'm closing this issue for now!
